lib/matplotlib/pyplot.py

```diff
@@ -2486,7 +2482,7 @@ def polar(*args, **kwargs) -> list[Line2D]:
 # is compatible with the current running interactive framework.
 if (rcParams["backend_fallback"]
         and rcParams._get_backend_or_none() in (  # type: ignore
-            set(rcsetup.interactive_bk) - {'WebAgg', 'nbAgg'})
```

Comment:
(Replying to https://github.com/matplotlib/matplotlib/pull/27719#discussion_r1478201234.)  The logic here is actually wrong:
```sh
MATPLOTLIBRC=<(printf 'backend: qtcairo') python -c 'from PyQt6.QtWidgets import *; app = QApplication([]); from pylab import *; print(gcf().canvas)'
```
currently shows that the matplotlibrc entry gets ignored and the qtagg backend gets selected (because of this line).

I think the proper logic here should be "if backend_fallback is on and _get_backend_or_none() returns a non-None backend *that is not compatible with the running interactive framework*, then fallback to the autoselection logic".  If one fixes that, then there won't be any need to special-case the web backends anymore (and thus INTERACTIVE_NON_WEB can be removed from this PR) as they are compatible with any interactive framework.

We don't have to block this PR on fixing this (side) issue; I would suggest just keeping the special-casing for webagg/nbagg here for now and open a separate issue to track this.

Reply:
That all makes sense to me.